include/mbedtls/bn_mul.h

```diff
@@ -192,6 +192,30 @@
 
 #endif /* AMD64 */
 
+#if defined(__aarch64__)
+
+#define MULADDC_INIT                \
+    asm(
+
+#define MULADDC_CORE                \
+        "ldr x4, [%3], #8   \n\t"   \
+        "ldr x5, [%4]       \n\t"   \
+        "mul x6, x4, %6     \n\t"   \
+        "umulh x7, x4, %6   \n\t"   \
+        "adds x5, x5, x6    \n\t"   \
+        "adc x7, x7, xzr    \n\t"   \
+        "adds x5, x5, %5    \n\t"   \
+        "adc %0, x7, xzr    \n\t"   \
+        "str x5, [%1], #8   \n\t"
+
+#define MULADDC_STOP                            \
+         : "+r" (c),  "=r" (d), "=r" (s)        \
```

Comment:
Apparently yes. I first had `=` but then the tests failed. Maybe it would be fine to only have the read-write output register but to not list it separately as an input register. I'm not sure if there's a difference in semantics.

Reply:
No difference that I'm aware of, although some old compilers might not support `+`. Verying between architectures, probably based on who wrote it, we've used `=` and repeated the register in both input and output lists, or used `+` and avoided repeating the register in the input list.